bvm/ballerina-runtime/src/main/java/org/ballerinalang/jvm/values/api/BValueCreator.java

```diff
@@ -0,0 +1,438 @@
+ import org.ballerinalang.jvm.values.DecimalValue;
+ import org.ballerinalang.jvm.values.ErrorValue;
+ import org.ballerinalang.jvm.values.FPValue;
+ import org.ballerinalang.jvm.values.StreamValue;
+ import org.ballerinalang.jvm.values.StreamingJsonValue;
+ import org.ballerinalang.jvm.values.TableValue;
+ import org.ballerinalang.jvm.values.TypedescValue;
+ import org.ballerinalang.jvm.values.XMLItem;
+ import org.ballerinalang.jvm.values.XMLQName;
+ import org.ballerinalang.jvm.values.XMLSequence;
+
+ import java.io.InputStream;
+ import java.math.BigDecimal;
+ import java.util.Map;
+ import java.util.function.Consumer;
+ import java.util.function.Function;
+
+ /**
+  * Helper class to create ballerina value instances.
+  *
+  * @since 1.1.0
+  */
+ public class BValueCreator {
+     /**
+      * Creates a new array with given array type.
+      *
+      * @param type the {@code BArrayType} object representing the type
+      * @return the new array
+      */
+     public static BArray createArrayValue(BArrayType type) {
+         return new ArrayValue(type);
+     }
+
+     /**
+      * Creates a new tuple with given tuple type.
+      *
+      * @param type the {@code BTupleType} object representing the type
+      * @return the new array
+      */
+     public static BArray createArrayValue(BTupleType type) {
+         return new ArrayValue(type);
+     }
+
+     /**
+      * Creates a new integer array.
+      *
+      * @param values initial array values
+      * @return integer array
+      */
+     public static BArray createArrayValue(long[] values) {
+         return new ArrayValue(values);
+     }
+
+     /**
+      * Creates a new boolean array.
+      *
+      * @param values initial array values
+      * @return boolean array
+      */
+     public static  BArray createArrayValue(boolean[] values) {
+         return new ArrayValue(values);
+     }
+
+     /**
+      * Creates a new byte array.
+      *
+      * @param values initial array values
+      * @return byte array
+      */
+     public static BArray createArrayValue(byte[] values) {
+         return new ArrayValue(values);
+     }
+
+     /**
+      * Creates a new float array.
+      *
+      * @param values initial array values
+      * @return float array
+      */
+     public static BArray createArrayValue(double[] values) {
+         return new ArrayValue(values);
+     }
+
+     /**
+      * Creates a new string array.
+      *
+      * @param values initial array values
+      * @return string array
+      */
+     public static BArray createArrayValue(String[] values) {
+         return new ArrayValue(values);
+     }
+
+     /**
+      * Create a new Ref value array.
+      *
+      * @param values initial array values
+      * @param type {@code BType} of the array. Do not use for primitive types.
+      * @return ref Value array
+      */
+     public static BArray createArrayValue(Object[] values, BType type) {
```

Comment:
Shall we change the parameter `BType type` to `BArrayType type`, in all array-related methods?